docs/JLBP-0018.md

```diff
@@ -22,7 +22,8 @@ There are a number of problems with shading:
   to also roll in the security fix).
 - Shading cannot be performed on types used in a library's own public API as return
   types or method arguments.
-- Service entries under `META-INF/services` can easily be messed up. They need
+- Shading often corrupts non-code metadata in META-INF/MANIFEST.mf and other locations.
```

Comment:
```suggestion
- Shading often corrupts non-code metadata in META-INF/MANIFEST.mf and other files.
```

I think you meant "files in other locations." This is the same as "other files".

Reply:
```suggestion
- Shading often corrupts non-code metadata in the files under `META-INF`, such as `META-INF/MANIFEST.mf` and `META-INF/services`
```

Reply:
locations is correct, not files. There can be locations inside a JAR file.